src/features/device-logs/lib/store.ts

```diff
@@ -193,6 +193,11 @@ function handleStreamingWrite(
 		try {
 			// Don't flush if the backend is reporting as unavailable
 			if (buffer.length && backend.available) {
+				const limit = ctx.retention_limit;
+				if (buffer.length > limit) {
+					// Ensure the buffer cannot be larger than the retention limit
+					buffer.splice(0, buffer.length - limit);
```

Comment:
so this will only send the latest `limit-many` logs. Do we don't care about the older logs as the buffer.length will be set to 0 after publish.